packages/core/lib/index.js

```diff
@@ -45,6 +45,19 @@ const createNewWallet = async (self) => {
 const restoreExistingWallet = async (self, dappShare) => {
   await initWeb3Auth(self, dappShare)
 
+  // If the user is not connected, we need to connect them first before doing the request
+  if (!await isConnected(self)) {
+    // Connect auth 2.0 account with web3Auth
+    await self.web3Auth.connectTo(WALLET_ADAPTERS.OPENLOGIN, {
+      loginProvider: "jwt",
+      extraLoginOptions: {
+        id_token: await self.authProvider.getIdToken(),
+        verifierIdField: "sub", // same as your JWT Verifier ID
+        domain: self.web3AuthConfig.domain,
+      },
+    });
```

Comment:
this looks exact copy with the code we have in the above function. If so we should move it to a separate function

Reply:
I can move it to `initWeb3Auth` since it's called along with it both times it's used